packages/manager/modules/config/.eslintrc.js

```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  parser: '@typescript-eslint/parser',
+  plugins: ['@typescript-eslint'],
+  extends: [
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+  ],
+  overrides: [
+    {
+      files: ['*.ts'],
+      rules: {
+        '@typescript-eslint/explicit-module-boundary-types': 'off',
+        '@typescript-eslint/no-empty-function': 'off',
+        'lines-between-class-members': 'off',
+        'import/extensions': 'off',
+        'no-use-before-define': 'off',
+        'no-shadow': 'off',
+        '@typescript-eslint/no-shadow': ['error'],
+      },
+    },
+  ],
```

Comment:
Is this a standard configuration ?

Reply:
@marie-j Actually, no. But these are rules that were either in conflict with the current eslint config in the manager or some eslint config we had that didn't make sense in typescript. So i configured it to either be remplaced or not trigger in typescript files.

For instance, i disabled 'no-shadow' of the eslint to activate the no-shadow rule of the ts instead as it didn't understand everything. So the rule is actually still active, it's just suited for ts.

I disabled 'import/extensions' because the current eslint config in the manager wanted me to have extensions for the imports since it couldn't infer ".ts" files. 

The 'no-use-before-define' is a rule that is activated by eslint, but it doesn't make sense in typescript files, it reports incorrect errors so i disabled it. Actually, now that i think about it, there is an equivalent rule in TS '@typescript-eslint/no-use-before-define' that should report the right errors. I'll add this one to throw an error to be as close as possible with our eslint config.

I disabled the rule '@typescript-eslint/no-empty-function' because we return an empty function in the code and didn't want to change the behavior for now as the goal is just to introduce typescript.

The rule '@typescript-eslint/explicit-module-boundary-types' is there to force you to write the return type of your functions etc. But in the most cases, TS can infer return types correctly. I can actually put this one on 'warning' to recall us to check for the return type. 

And finally, the rule 'lines-between-class-members' was activated by our current eslint, but since now  we're implemeting interfaces in our classes, i thought that there were too many spaces. I can reactivate this one since it's the only one i disabled on personal preference !  

If you have other workarounds for some of these rules i'll take them into account, or if you want me to reactivate some of them i can too!

Reply:
@zakariasahmane Thank you for this verbose explanation :) I was just curious if this was some standard eslint config for TS or just adapted to our current context 
About `lines-between-class-members` if it is consistent with the rest of our codebase I don't mind keeping it